BetaUtils.Core/extensions/IEnumerableExtensions.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+
+namespace BetaUtils.Core.Extensions
+{
+    public static class IEnumerableExtensions
+    {
+        /// <summary>
+        /// Checks if IEnumerable is null or empty
+        /// </summary>
+        /// <typeparam name="items">IEnumerable</typeparam>
+        /// <returns>Bool</returns>
+        public static bool IsNullOrEmpty<T>(this IEnumerable<T> items)
+        {
+            return !items.Any();
+        }
+
+        /// <summary>
+        /// Iterates through IEnumerable
+        /// </summary>
+        /// <typeparam name="items">IEnumerable</typeparam>
+        /// <typeparam name="action">Action</typeparam>
+        /// <returns>IEnumerable</returns>
+        public static IEnumerable<T> ForEach<T>(this IEnumerable<T> items, Action<T> action)
+        {
+            if (action is null)
+            {
+                throw new ArgumentNullException(nameof(action));
+            }
+
+            foreach (var item in items)
+            {
+                action(item);
+            }
+
+            return items;
+        }
+
+        /// <summary>
+        /// Converts an IEnumerable to a HashSet
+        /// </summary>
+        /// <typeparam name="items">IEnumerable</typeparam>
+        /// <returns>HashSet</returns>
+        public static HashSet<T> ToHashSet<T>(this IEnumerable<T> items)
+        {
+            HashSet<T> hashSet = new HashSet<T>();
+
+            foreach (var item in items)
+            {
+                hashSet.Add(item);
+            }
+
+            return hashSet;
+        }
+
+        /// <summary>
+        /// Converts an IEnumerable to an Observable Collection
+        /// </summary>
+        /// <param name="items">IEnumerable</param>
+        /// <returns>ObservableCollection</returns>
+        public static ObservableCollection<T> ToObservableCollection<T>(this IEnumerable<T> items)
+        {
+            var observableCollection = new ObservableCollection<T>();
+
+            foreach (var item in items)
+            {
+                observableCollection.Add(item);
+            }
+
+            return observableCollection;
+        }
+
+        /// <summary>
+        /// Takes a collection and splits it into a batch of collections with provided number of values
+        /// </summary>
+        /// <param name="items">IEnumerable</param>
+        /// <param name="numberOfItemsPerBatch">Integer</param>
+        /// <returns>New IEnumerable collections split into collections with provided number of items</returns>
+        public static IEnumerable<IEnumerable<T>> Batch<T>(this IEnumerable<T> items, int numberOfItemsPerBatch)
+        {
+            return items.Select((item, totalNumber) => new { item, totalNumber })
```

Comment:
I got confused by name of this variable. If I got it right, this is just an index. I would suggest to name this variable in way that will suggest that meaning (like: index, order, orderInTotal...).
Please correct me if I got this wrong :)

Reply:
You are right, my mistake.
It represents the index of the current item in the original sequence, so it should be renamed.